File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.java
class Solution {
    public int maxSum (int [] nums) {
        int freq [] = new int [201];    
        int result = 0, maxNum = Integer.MIN_VALUE;
        for (int i = 0; i < nums.length; ++i) {
            freq [nums [i] + 100]++;
            if (nums [i] > maxNum) maxNum = nums [i];
        }
        for (int i = 0; i < 201; ++i){
            int num = i - 100;
            if (freq [i] > 0 && num > 0)
                result += num;
        }
        if (result != maxNum && result == 0)
            return maxNum;
        return result;
    }
}

File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.py
class Solution:
    def maxSum (self, nums : List[int]) -> int:
        freq = [0] * 201
        result = 0; max_num = float ('-inf')
        for i in range (len (nums)):
            freq [nums [i] + 100] += 1
            if nums [i] > max_num: max_num = nums [i]
        for i in range (201):
            num = i - 100
            if freq [i] > 0 and num > 0: result += num
        if result != max_num and result == 0:
            return max_num
        return result
